packages/core-utils/src/route.js

```diff
@@ -1,31 +1,151 @@
  */
-export function getTransitOperatorFromId(feedId, transitOperators) {
+export function getTransitOperatorFromFeedIdAndAgencyId(
+  feedId,
+  agencyId,
+  transitOperators
+) {
   return (
-    transitOperators.find(transitOperator => transitOperator.id === feedId) ||
-    null
+    transitOperators.find(
+      transitOperator =>
+        transitOperator.feedId === feedId &&
+        transitOperator.agencyId === agencyId
+    ) || null
   );
 }
 
-/** Looks up an operator from the provided configuration */
+/**
+ * Looks up an operator from the provided leg.
+ *
+ * @param  {object} leg The Itinerary Leg from which to find the transit
+ *    operator
+ * @param  {object} transitOperators transitOperators from config.
+ * @return {object} the operator if one was found or null if no match was found
+ */
+export function getTransitOperatorFromLeg(leg, transitOperators) {
+  if (!leg.routeId || !leg.agencyId) return null;
+  const feedId = leg.routeId.split(":")[0];
+  return getTransitOperatorFromFeedIdAndAgencyId(
+    feedId,
+    leg.agencyId,
+    transitOperators
+  );
+}
+
+/**
+ * Looks up an operator from the provided configuration given an OTP route.
+ * NOTE: this assumes the use of the OTP Route model or a modified OTP
+ * RouteShort model (such as the one found in the IBI fork of OTP) that also
+ * returns the agencyId.
+ *
+ * @param  {object} route Either an OTP Route or RouteShort model
+ * @param  {array} transitOperators transitOperators from config
+ * @return {object} the operator if one was found or null if no match was found
+ */
 export function getTransitOperatorFromOtpRoute(route, transitOperators) {
   if (!route.id) return null;
-  const operatorId = route.id.split(":")[0].toLowerCase();
-  return getTransitOperatorFromId(operatorId, transitOperators);
+  const feedId = route.id.split(":")[0];
+  let agencyId;
+  if (route.agency) {
+    // This is returned in the OTP Route model
+    agencyId = route.agency.id;
+  } else if (route.agencyId) {
+    // This is returned in the OTP RouteShort model (such as in the IBI fork)
+    agencyId = route.agencyId;
+  } else {
+    return null;
+  }
+  return getTransitOperatorFromFeedIdAndAgencyId(
+    feedId,
+    agencyId,
+    transitOperators
+  );
 }
 
+// The functions below are for enhanced route sorting functions for the route
+// viewer on OTP-react-redux.
+// They address route ordering issues discussed in
+// https://github.com/opentripplanner/otp-react-redux/pull/123 and
+// https://github.com/opentripplanner/otp-react-redux/pull/124.
+
 /**
- * The functions below are for enhanced route sorting functions for
- * the route viewer on OTP-react-redux.
- * They address route ordering issues discussed in
- * https://github.com/opentripplanner/otp-react-redux/pull/123 and
- * https://github.com/opentripplanner/otp-react-redux/pull/124.
+ * A large comparator value that can safely be used in mathematical sort
+ * comparisons to place things at the ned of lists
```

Comment:
typo "ned"

Reply:
Fixed in 5163a219c2ab5072d9338041435293fda8877c42